app/models/edition.rb

```diff
@@ -245,6 +245,12 @@ def self.only_broken_links
     )
   end
 
+  def self.review_overdue
+    joins("INNER JOIN documents ON documents.id = editions.document_id INNER JOIN review_reminders ON review_reminders.document_id = documents.id")
+    .where(document: { review_reminders: { review_at: ..Time.zone.today } })
+    .where.not(first_published_at: nil)
+  end
+
```

Comment:
Have just tested this and it seems to produce the correct SQL 
```
  scope :review_overdue, -> {
    joins(document: :review_reminder)
      .where(review_at: ..Time.zone.today)
      .where.not(first_published_at: nil)
  }
```

Produces
```
"SELECT * FROM `editions` 
INNER JOIN `documents` ON `documents`.`id` = `editions`.`document_id` 
INNER JOIN `review_reminders` ON `review_reminders`.`document_id` = `documents`.`id` 
WHERE `editions`.`state` != 'deleted' 
AND `editions`.`review_at` <= '2023-09-04' 
AND `editions`.`first_published_at` IS NOT NULL"
```

Reply:
I think it might be a bit more readable? what do you think?
I also feel like it should be a scope rather than a self method to accurately portray what the method is doing

Reply:
hmm so interestingly if i use this scope 

```
  scope :review_overdue, -> {
    joins(document: :review_reminder)
      .where(document: { review_reminders: { review_at: ..Time.zone.today } })
      .where.not(first_published_at: nil)
  }
```

my output is 

```
 FROM `editions` INNER JOIN `documents` `document` ON `document`.`id` = `editions`.`document_id` INNER JOIN `review_reminders` ON `review_reminders`.`document_id` = `document`.`id` WHERE `editions`.`state` != 'deleted' AND `review_reminders`.`review_at` <= '2023-09-04' AND `editions`.`first_published_at` IS NOT NULL"
```
It's interesting that mine is aliasing the table to `document`.

Due to that, it's causing this error

https://github.com/alphagov/whitehall/assets/42515961/a7041702-5564-49bb-a46b-a7c87408aba6

I'd be interested to see if that bug is replicable for you if you get a bit of time please 🙏 

steps to recreate would be

1. create a review reminder where the review date has passed (probs easiest from console)
2. search for it by title/slug while the review overdue filter is checked

Def agreed that using a scope is way nicer 👍 I'll sort that now